mm2src/coins_activation/src/tendermint_with_assets_activation.rs

```diff
@@ -187,18 +235,27 @@ impl PlatformCoinWithTokensActivationOps for TendermintCoin {
     ) -> Result<Self, MmError<Self::ActivationError>> {
         let conf = TendermintConf::try_from_json(&ticker, coin_conf)?;
 
-        let priv_key_build_policy =
-            PrivKeyBuildPolicy::detect_priv_key_policy(&ctx).mm_err(|e| TendermintInitError {
-                ticker: ticker.clone(),
-                kind: TendermintInitErrorKind::Internal(e.to_string()),
-            })?;
+        let activation_policy = if let Some(pubkey) = activation_request.with_pubkey {
+            if ctx.is_watcher() || ctx.use_watchers() {
+                return MmError::err(TendermintInitError {
+                    ticker: ticker.clone(),
+                    kind: TendermintInitErrorKind::CantUseWatchersWithPubkeyPolicy,
+                });
```

Comment:
Can you give a little bit more context about watchers using external wallet?

Reply:
It's just another signed transaction that gets sent to the watcher where they can add the secret once revealed by the maker, so it will be one more transaction signed by keplr. Maybe for Tendermint watchers support, we will need some additions to nucleus codebase, I need to check this but it's not a priority currently.

Reply:
> If I remember correctly, we made watchers so people don't have to keep the mm2 open for the swaps. If user has to be online/active on the keplr window, how does using watcher makes sense?

The main reason we need watchers is not offline swaps but to fix some security issues, if the taker goes offline after sending their payment for a long time the maker can get both payments. Watchers start their work after the taker sends their payment, so for keplr the taker will sign 2 transactions, one for the taker payment and another to send to the watchers, then if anything happens and they went offline, the watcher can complete the swap or refund it before the maker refunds theirs.

Reply:
P.S. if a swap fails, the taker has to wait some time before they can refund it. If they didn't get back online after some time to refund it, and the maker is malicious, they can lose their payment.